pkg/util/iptables/iptables.go

```diff
@@ -783,3 +802,75 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+type ParseError interface {
+	Line() int
+	Error() error
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+const (
+	colon = ":"
+)
+
+func (e parseError) Error() error {
+	return fmt.Errorf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// parseRestoreError extracts the line from the error, if it matches returns parseError
+// for example:
+// input: iptables-restore: line 51 failed
+// output: parseError:  cmd = iptables-restore, line = 51
+// NOTE: parseRestoreError depends on the error format of iptables, if it ever changes
+// we need to update this function
+func parseRestoreError(err string) (ParseError, bool) {
+	errInfos := strings.SplitN(err, colon, 3)
+	if len(errInfos) != 3 {
+		return nil, false
+	}
+
+	cmdInfo := strings.TrimSpace(errInfos[1])
+	lineInfos := strings.Fields(errInfos[2])
+	if len(lineInfos) != 3 {
+		return nil, false
+	}
+
+	line, errMsg := strconv.Atoi(lineInfos[1])
+	if errMsg != nil {
+		return nil, false
+	}
+	return parseError{cmd: cmdInfo, line: line}, true
+}
+
+// ExtractLines extracts the -count and +count data from the lineNum row of lines and return
+// NOTE: lines start from line 0
+func ExtractLines(limit, lineNumber int, lines []byte) []string {
```

Comment:
I think the args should be `[]byte, line, context ` or just `[]byte, line, count` (let the caller do the context math)

Reply:
caller calculates start and end?